v5migrationhelper/orm/src/main/java/org/hibernate/search/query/hibernate/impl/FullTextQueryImpl.java

```diff
@@ -222,12 +222,14 @@ public FullTextQueryImpl setSpatialParameters(double latitude, double longitude,
 	@Override
 	public FullTextQuery setMaxResults(int maxResults) {
 		super.setMaxResults( maxResults );
+		hSearchQuery.maxResults( maxResults );
```

Comment:
I don't know how this could fix your problem, but one thing is for sure: this is not the right way of propagating this parameter. People can set the max results through other means (e.g. `getOptions().getLimit().setMaxRows()`). So we *have* to call hSearchQuery.maxResults() before every single query execution, just in case.

Reply:
I'll apply the fix we discussed instead, to avoid passing `Integer.MAX_VALUE` to the backend.